platforms/documentation/docs/src/docs/userguide/dep-man/declaring-repositories/supported_repository_types.adoc

```diff
@@ -0,0 +1,160 @@
+
+[[sub:flat_dir_resolver]]
+== Flat directory repository
+
+Some projects store dependencies on a shared drive or within the project’s source code rather than using a binary repository.
+To use a flat filesystem directory as a repository, you can configure it like this:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=flat-dir-multi]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=flat-dir-multi]"]
+====
+
+This configuration adds repositories that search specified directories for dependencies.
+
+Flat directory repositories do not support metadata formats like Ivy XML or Maven POM files.
+Instead, Gradle dynamically generates a module descriptor based on the presence of artifacts, without any dependency information.
+
+Gradle prefers modules with real metadata over those generated by flat directory repositories, so flat directories cannot override artifacts with real metadata from other declared repositories.
+For instance, if Gradle finds `jmxri-1.2.1.jar` in a flat directory and `jmxri-1.2.1.pom` in another repository, it will use the metadata from the latter.
+
+If you need to override remote artifacts with local ones, consider using an Ivy or Maven repository pointing to a local directory instead.
+
+[[sec:maven_repo]]
+== Maven repositories
+
+Many organizations host dependencies in private Maven repositories accessible within their network.
+Gradle can declare Maven repositories by specifying their URL:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=maven-like-repo]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=maven-like-repo]"]
+====
+
+[[sub:custom-maven-repo]]
+=== Composite Maven repository
+
+Sometimes, POMs are published in one location, and JARs in another.
+You can define such a repository as follows:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=maven-like-repo-with-jar-repo]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=maven-like-repo-with-jar-repo]"]
+====
+
+Gradle will first look for POMs and artifacts at the base URL, and if the artifact is not found, it will check the additional `artifactUrls`.
+
+[[sub:accessing_secured_maven_repositories]]
+=== Authenticated Maven repository
+
+You can specify credentials for Maven repositories that require authentication.
+See <<supported_repository_protocols.adoc#supported_transport_protocols>> for authentication options.
+
+[[sub:maven_local]]
+=== Local Maven repository
+
+Gradle can also consume dependencies from the link:https://maven.apache.org/guides/introduction/introduction-to-repositories.html[local Maven repository].
+This is useful for teams that publish locally with one project and consume the artifacts with another.
+To declare the local Maven cache as a repository:
```

Comment:
```suggestion
Gradle can also consume dependencies from the link:https://maven.apache.org/guides/introduction/introduction-to-repositories.html[local Maven repository]:
```

This pattern should really be discouraged and is only really intended for integration with Maven.

Reply:
Simplified and discouraged.